### PR TITLE
Add rollback test after migration

### PR DESCRIPTION
**Pull request by mitiao**
poo#17836
Introduce a new var ROLLBACK_AFTER_MIGRATION to implement migraiton and its rollback in 1 testcase and control the workflow.
It would not break existing relevant tests and retain 2 tests mode for migration and rollback.
Also add rollback test for ZDUP migration.

Verified run:
offline migration and rollback: http://147.2.207.208/tests/440
online migration and rollback: http://147.2.207.208/tests/442
zdup migration and rollback: http://147.2.207.208/tests/447
boot_to_snapshot: http://147.2.207.208/tests/443
rollback from hdd image: http://147.2.207.208/tests/444

### CODE REVIEW

**Comment by mitiao** [comment]
travis want to kill main.pm?

File: products/sle/main.pm [hunk 1]
```diff
@@ -937,6 +939,15 @@ sub prepare_target() {
...
 
+sub load_rollback_tests() {
+    loadtest "boot/grub_test_snapshot";
+    if (get_var('UPGRADE') || get_var('ZDUP')) {
+        loadtest "boot/snapper_rollback";
+    }
+    if (get_var('MIGRATION_ROLLBACK')) {
+        loadtest "online_migration/sle12_online_migration/snapper_rollback";
+    }
+
```
Comment (okurz) [inline products/sle/main.pm]: while you are at it. It would be great if you can crosscheck with the opensuse case. Especially update/upgrade on openSUSE Leap is getting more and more important ( @lnussel )

File: tests/boot/grub_test_snapshot.pm [hunk 2]
```diff
@@ -16,14 +16,24 @@ use base "basetest";
 use testapi;
 
 sub run() {
-    assert_screen "grub2";
+    if (get_var('ROLLBACK_AFTER_MIGRATION')) {
+        # set BOOT_TO_SNAPSHOT here rather than in main.pm to prevent conflict with existing snapshot tests
+        set_var('BOOT_TO_SNAPSHOT', 1);
+        select_console 'root-console';
+        wait_still_screen 2;
```
Comment (okurz) [inline tests/boot/grub_test_snapshot.pm]: that should not be necessary anymore after the implicit assert_screen in select_console now

**Comment by okurz** [comment]
> travis want to kill main.pm?

When tidy wants to change indention on anything probably you have a syntax error, missing `;` or such

File: tests/boot/grub_test_snapshot.pm (hunk 2 above)
Comment (mitiao) [inline tests/boot/grub_test_snapshot.pm]: removed

File: products/sle/main.pm (hunk 1 above)
Comment (mitiao) [inline products/sle/main.pm]: need time to crosscheck and verification. may i send another PR for it? :)

File: products/sle/main.pm [hunk 3]
```diff
@@ -769,7 +769,9 @@ sub load_x11tests() {
         loadtest "x11/reboot_gnome";
     }
     loadtest "x11/desktop_mainmenu";
-    loadtest "x11/shutdown";
+    unless (get_var('ROLLBACK_AFTER_MIGRATION')) {
```
Comment (okurz) [inline products/sle/main.pm]: I guess it's not quite clear why you want to exclude the shutdown here. maybe a short comment?

File: products/sle/main.pm (hunk 1 above)
Comment (okurz) [inline products/sle/main.pm]: sure

File: products/sle/main.pm (hunk 3 above)
Comment (mitiao) [inline products/sle/main.pm]: sure. comment added.
